specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json

```diff
@@ -3562,6 +3562,408 @@
+          "ThreatIntelligence"
+        ],
+        "description": "Delete a threat intelligence.",
+        "operationId": "ThreatIntelligenceIndicator_Delete",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/OperationalInsightsResourceProvider"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "#/parameters/ThreatIntelligenceIdentifier"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+          "204": {
+            "description": "No Content"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/threatIntelligence/indicators/query": {
+      "post": {
+        "x-ms-examples": {
+          "Get all threat intelligence.": {
+            "$ref": "./examples/threatintelligence/QueryThreatIntelligence.json"
+          }
+        },
+        "tags": [
+          "ThreatIntelligence"
+        ],
+        "description": "Query all threat intelligence.",
+        "operationId": "ThreatIntelligenceIndicatorsList_Query",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/OperationalInsightsResourceProvider"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "#/parameters/ThreatIntelligenceArmStixQuery"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ThreatIntelligenceResourceList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/threatIntelligence/indicators/metrics": {
```

Comment:
If this is a "GET" it needs to return a list of Microsoft.SecurityInsights/threatIntelligence/metrics" resources according to the URI structure standards I outlined above.  If you change this to a POST with the action name "listMetrics" or similar you can keep the API as is

Reply:
I see. 
If we move to POST threatIntelligence/metrics/listMetrics then metrics is {resourceType}. 
if we leave it as GET threatIntelligence/metrics, we need to return a collection of metrics since metrics is {resourceType}. This will require us to make a lot of change. 
Why dont we consider **metrics** as a {resourceName} of threatintelligence {resourceType} ? We are return a ThreatIntelligenceMetric object anyway

Reply:
Like I said in a previous comment. threatintellgience is a singleton, and we could have a collection of metrics exposed by this singleton, in which case GET threatintelligence/metrics (or threatintelligence/indicatorMetrics) is correct.

Reply:
This URI structure is OK if you return a valid collection response `{ "value": [{<entity1>, <entity2>}] }` and if each entity in that response has a compliant resource body (i.e. top level "name", "id", and "type")

Reply:
Changing the return type could be an issue for us. We can change this URI to POST Microsoft.SecurityInsights/threatIntelligence/metrics/listMetrics and keep the response format